charts/images.yaml

```diff
@@ -53,7 +53,7 @@ images:
 - name: csi-provisioner
   sourceRepository: github.com/kubernetes-csi/external-provisioner
   repository: k8s.gcr.io/sig-storage/csi-provisioner
-  tag: "v2.1.2"
+  tag: "v3.0.0"
```

Comment:
First, I have the same comment such as https://github.com/gardener/gardener-extension-provider-gcp/pull/321/files#r704090886:

> Update external-snapshotter dependency to v4, enabling use of the volume snapshot v1 client, which requires minimum Kubernetes version 1.20 (#650, @xinyanw409)

Second, wrt to updating to `external-snapshotter@v4` I have the following concern. In https://github.com/gardener/gardener-extension-provider-aws/pull/366#issue-930985490 I described that `external-snapshotter@v3` features a new validating webhook component. We currently don't install this new validating webhook component. The note from the external-snapshotter REAME.md is exactly this one:

> :warning: **WARNING**: Cluster admins choosing not to install the webhook server and participate in the phased release process can cause future problems when upgrading from `v1beta1` to `v1` volumesnapshot API, if there are currently persisted objects which fail the new stricter validation. Potential impacts include being unable to delete invalid snapshot objects.

`external-snapshotter@v4` is exactly the version that upgrades the API from `v1beta1` to `v1`. So before upgrading to `external-snapshotter@v4` we have to decide what we do with the external-snapshotter validating webhook component - do we install it; do we proceed without it (highly not recommended by the upstream project) and what would be the impact/consequences then? Without clarifying these questions, there is no point to update to the affected version of the sidecars.

/cc @rfranzke 
/hold

Reply:
Good points @ialidzhikov, I'd vote to follow the suggestion of the community and install this webhook to make sure that future problems are prevented/minimized.

Reply:
Hi,
I added the fix regarding targetVersion.

If you decide to install the webhook, will it be ok that I'll create the separate pr for it?

Reply:
And if yes,
We have an architecture question:
Should the validation webhook be configured in the seed cluster or in the shoot cluster?

Reply:
The webhook server should run in the seed cluster targeting the shoot (similar to how the csi-controllers are ran). Maybe it might even make sense to put it as a sidecar container into the csi-snapshot-controller pod, not sure what the community recommends here.

Reply:
@rfranzke,
Do you know where the resources (Deployment/Service/etc.) of the csi-controllers are configured in the seed cluster? In which repo can I find these resources?
Also, can I get access to seed cluster to see how the resources are implemented in the cluster?

Reply:
https://github.com/gardener/gardener-extension-provider-aws/tree/master/charts/internal/seed-controlplane/charts/csi-driver-controller/templates

> Also, can I get access to seed cluster to see how the resources are implemented in the cluster?

I don't get this question, you can run Gardener with this extension and your own seeds. There are no "central seed clusters" one could get access to.